### ISSUE TITLE
flux kvs dir -R doesn't show empty directories

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#1449]
I just stumbled across this:
```
$ flux kvs mkdir a.b.c
$ flux kvs dir -R a.b
$ flux kvs dir a.b
a.b.c.
```

This seems surprising and nothing in flux-kvs(1) prepares one for that result:

> **dir** [-R] [-d] [-w COLS] [-a treeobj] [key]
           Display all keys and their values under the directory key. If key
           does not exist or is not a directory, display an error message. If
           key is not provided, "." (root of the namespace) is assumed. If -R
           is specified, recursively display keys under subdirectories. If -d
           is specified, do not output key values. Output is truncated to fit
           the terminal width.  -w COLS sets the terminal width (0=unlimited).
           -a treeobj causes the lookup to be relative to an RFC 11 snapshot
           reference.

**Comment by chu11** [comment]
Interesting.  It does seem to just be a corner case:

```
        } else if (flux_kvsdir_isdir (dir, name)) {                                                                                    
            if (Ropt) {                                                                                                                
                const flux_kvsdir_t *ndir;                                                                                             
                if (!(f = flux_kvs_lookupat (h, FLUX_KVS_READDIR, key, rootref))                                                       
                        || flux_kvs_lookup_get_dir (f, &ndir) < 0)                                                                     
                    log_err_exit ("%s", key);                                                                                          
                dump_kvs_dir (ndir, maxcol, Ropt, dopt);                                                                               
                flux_future_destroy (f);                                                                                               
            } else                                                                                                                     
                printf ("%s.\n", key);                  
```

If there is nothing, the recursive call does nothing.

You think it should output the directory just like the above "else" case?

**Comment by garlick** [comment]
I think so, what do you think?

After I posted I had some doubts what the "value" should be but maybe just like without -R makes sense.

**Comment by chu11** [comment]
Certainly the inconsistency bothers me.

I'm actually leaning towards removing output of directories in its entirety (i.e. remove the else statement).  This would be consistent with the documentation of ```Display all keys and their values under the directory key.```  We could add a note in the documentation about how empty directories won't be shown.  This is also consistent to why we decided to leave the ```dir``` command when the ```ls``` command was added.  We wanted a way to show all the values in a directory.

The alternate would be to always output the directory before recursing, like:

```
            printf ("%s.\n", key);
            if (Ropt) {                                                                                                                
                  /* recurse */
            }          
```

so that the directory is consistently shown.
